src/main/java/org/bitlet/wetorrent/bencode/Bencode.java

```diff
@@ -100,16 +97,18 @@ private void print(Object object, OutputStream os) throws IOException {
         }
     }
 
-    private Object parse(InputStream is) throws IOException {
-        is.mark(0);
-        int readChar = is.read();
-        switch (readChar) {
+    private int readHead(InputStream is) throws IOException {
```

Comment:
We can avoid this method and just call is.read in every place instead of readHead(is).